types/react-router-navigation/index.d.ts

```diff
@@ -0,0 +1,182 @@
+// Type definitions for react-router-navigation 1.0
+// Project: https://github.com/LeoLeBras/react-router-navigation#readme
+// Definitions by: Kalle Ott <https://github.com/kaoDev>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.4
+
+import { Component, ReactNode, ReactElement, ComponentClass } from "react";
+import { StyleProp, ViewProperties, ViewStyle, TextStyle } from "react-native";
+import { TabViewAnimated, TabViewPagerPan } from "react-native-tab-view";
+import { RouteProps } from "react-router-navigation-core";
+import {
+    NavigationTransitionProps,
+    NavigationTransitionSpec
+} from "react-navigation";
+
+/**
+ * Navigation
+ */
+
+export type CardProps = RouteProps & NavBarProps;
+
+export type CardSubViewProps = any;
+//  NavigationSceneRendererProps &
+//   CardsRendererProps &
+//   CardProps
+
+export type NavBarProps = {
+    // General
+    hideNavBar?: boolean;
+    renderNavBar?: (props: CardSubViewProps) => ReactNode;
+    navBarStyle?: StyleProp<ViewStyle>;
+    // Left button
+    hideBackButton?: boolean;
+    backButtonTintColor?: string;
+    backButtonTitle?: string;
+    renderLeftButton?: (props: CardSubViewProps) => ReactNode;
+    // Title
+    title?: string;
+    titleStyle?: StyleProp<TextStyle>;
+    renderTitle?: (props: CardSubViewProps) => ReactNode;
+    // Right button
+    renderRightButton?: (props: CardSubViewProps) => ReactNode;
+};
+
+export type NavigationProps = NavBarProps & {
+    cardStyle?: StyleProp<ViewStyle>;
+    configureTransition?: (
+        transitionProps: NavigationTransitionProps,
+        prevTransitionProps?: NavigationTransitionProps
+    ) => NavigationTransitionSpec;
+    onTransitionStart?: (...args: any[]) => void;
+    onTransitionEnd?: (...args: any[]) => void;
+};
+
+export type Card = CardProps & { key: string };
+
+/**
+ * Tabs
+ */
+
+export type TabSubViewProps = any;
+//  SceneRendererProps &
```

Comment:
Why use `any` and not the intersection?

Reply:
because it is taken directly from the flow-types of the package. I hope those types will get finalized when the lib gets beyond RC state